Sources/NIO/ChannelHandler.swift

```diff
@@ -33,7 +33,12 @@ public protocol ChannelHandler: class {
 
 /// Untyped `ChannelHandler` which handles outbound I/O events or intercept an outbound I/O operation.
 ///
-/// We _strongly_ advice against implementing this protocol directly. Please implement `ChannelOutboundHandler`.
+/// Despite the fact that `write` is one of the methods on this `protocol`, you should avoid assuming that "outbound" events are to do with
+/// writing to sockets. Instead, "outbound" events are events that are passed *to* the socket (or other event source): that is, things you tell
+/// the socket to do. That includes `write` ("write this data to the socket"), but it also includes `read` ("please begin attempting to read from
+/// the socket") and `bind` ("please bind the following address"), which have nothing to do with sending data.
```

Comment:
@Lukasa I am not sure we should use `socket` here as a `Channel` may not be backed by a `socket`.

Reply:
That said the rest of the docs sounds good... Maybe just make it clear that `socket` is an example here.